### ISSUE TITLE
Deprecated functions in Asterisk v21

### ISSUE THREAD

**Issue by 2br-2b** [marcelloceschia/chan-sccp-b#609]
While I can't find the exact documentation, ast_channel_macroexten is deprecated in Asterisk v21. The sources I have for this are I-tried-it and this commit: https://github.com/asterisk/asterisk/commit/e8f548c1557620df8a7231cbede6cb9b052d352b

I haven't confirmed that it still works on v20, but it looks like that commit is only in the v21 tags, so I assume it still works fine there

**Reproduction Steps:**

1. Try to install chan-sccp on Asterisk v21
2. Get an error

**chan-sccp version:** v4.3.5

**asterisk version:** v21+

**Comment by lachesis** [comment]
Confirmed, doesn't build for me on Asterisk 21 either. I guess the right answer is to remove all of the macro related code?

**Comment by lachesis** [comment]
It built for me with this patch:
```
diff --git a/src/pbx_impl/ast116/ast116.c b/src/pbx_impl/ast116/ast116.c
index c162e755..f6364dca 100644
--- a/src/pbx_impl/ast116/ast116.c
+++ b/src/pbx_impl/ast116/ast116.c
@@ -3065,10 +3065,6 @@ DECLARE_PBX_CHANNEL_STRGET(name)
     DECLARE_PBX_CHANNEL_STRGET(linkedid)
     DECLARE_PBX_CHANNEL_STRGET(context)
     DECLARE_PBX_CHANNEL_STRSET(context)
-    DECLARE_PBX_CHANNEL_STRGET(macroexten)
-    DECLARE_PBX_CHANNEL_STRSET(macroexten)
-    DECLARE_PBX_CHANNEL_STRGET(macrocontext)
-    DECLARE_PBX_CHANNEL_STRSET(macrocontext)
     DECLARE_PBX_CHANNEL_STRGET(call_forward)
     DECLARE_PBX_CHANNEL_STRSET(call_forward)

@@ -3536,10 +3532,6 @@ const PbxInterface iPbx = {
        setChannelExten: sccp_astwrap_set_channel_exten,
        getChannelContext: sccp_astwrap_get_channel_context,
        setChannelContext: sccp_astwrap_set_channel_context,
-       getChannelMacroExten: sccp_astwrap_get_channel_macroexten,
-       setChannelMacroExten: sccp_astwrap_set_channel_macroexten,
-       getChannelMacroContext: sccp_astwrap_get_channel_macrocontext,
-       setChannelMacroContext: sccp_astwrap_set_channel_macrocontext,
        getChannelCallForward: sccp_astwrap_get_channel_call_forward,
        setChannelCallForward: sccp_astwrap_set_channel_call_forward,

@@ -3687,10 +3679,6 @@ const PbxInterface iPbx = {
        .setChannelExten = sccp_astwrap_set_channel_exten,
        .getChannelContext = sccp_astwrap_get_channel_context,
        .setChannelContext = sccp_astwrap_set_channel_context,
-       .getChannelMacroExten = sccp_astwrap_get_channel_macroexten,
-       .setChannelMacroExten = sccp_astwrap_set_channel_macroexten,
-       .getChannelMacroContext = sccp_astwrap_get_channel_macrocontext,
-       .setChannelMacroContext = sccp_astwrap_set_channel_macrocontext,
        .getChannelCallForward = sccp_astwrap_get_channel_call_forward,
        .setChannelCallForward = sccp_astwrap_set_channel_call_forward,
```
The module that it build loaded. I have not tested further yet. Will report back once my first SCCP phone is here and I test it out later today.

**Comment by lachesis** [comment]
Reporting back as promised. I've been running with this simple patch for over a month now, with no major issues. This probably breaks macro support on older Asterisk. I would guess the correct answer is to add some new entries to the `pbx_impl` then do the right autotools magic to make it select those for 21+.

**Comment by endor3821** [comment]
has there been any development on this issue. perhaps a ./configure --disable-macros feature

**Comment by 2br-2b** [comment]
I don't believe the original contributors are still active at this time. If you build from the PR branch, that should build with the proposed changes

**Comment by endor3821** [comment]
So after a lot of research, I have discovered the issue. The sccp protocol is being phased out. There may be many reasons for this, ie cisco complaints against open source use of it. Asterisks >19 has dropped the chan_skinny and ast_macro from the official build. So the only way to truly fix this is to recompile asterisk with these modules. This is a daunting task seeing has it most likely will break other portions of the code body. So it wont be long until these devices are relegated to the past.

**Comment by fhloston** [comment]
so far chan-sccp also compiles with asterisk 22 with these [1] changes. phones register fine.

however, during an attempted call from pjsip i do get the error "Don't know how to send format alaw packets with RTP" from this issue [2].

same for the echotest application.

@lachesis have you actually been able to route calls  to chan_sccp devices/channels? what am i missing?
 
[1] https://github.com/chan-sccp/chan-sccp/issues/609#issuecomment-1950275587
[2] https://github.com/chan-sccp/chan-sccp/issues/617

**Comment by MadManChLP** [comment]
are ther any new informations or maby alternativs to chan-sccp becaws i realy would like to implement cisco phones into FfreePBX 17 witch ships with asterisk 22.5.2 by default

**Comment by ibuntumain** [comment]
@MadManChLP same i have a set of 7965's and will register for a frew minutes then not register anymore (using SIP)
